File: algorithm/src/main/java/baekjoon/p05671/Main.java
package baekjoon.p05671;

import java.util.Scanner;

public class Main {

    public static void main(String[] args) {
        Scanner sc = new Scanner(System.in);
        while (sc.hasNext()) {
            System.out.println(solve(sc.nextInt(), sc.nextInt()));
        }
    }

    private static int solve(int n, int m) {
        int ret = 0;
        for (int i = n; i <= m; i++) {
            if (check(i)) {
                ret++;
            }
        }

        return ret;
    }

    private static boolean check(int n) {
        int[] cnt = new int[10];
        while (n != 0) {
            cnt[n % 10]++;
            n /= 10;
        }

        for (int e : cnt) {
            if (e > 1) {
                return false;
            }
        }

        return true;
    }

}

File: algorithm/src/main/java/baekjoon/p05671/Main.kt
package baekjoon.p05671

fun main(args: Array<String>) {
    while (true) {
        val input = readLine() ?: break
        val inputs = input.split(" ").map { it.toInt() }
        println(solve(inputs[0], inputs[1]))
    }
}

private fun solve(n: Int, m: Int): Int {
    return (n..m).count { check(it) }
}

private fun check(n: Int): Boolean {
    var nn = n
    val cnt = IntArray(10)

    while (nn != 0) {
        cnt[nn % 10]++
        nn /= 10
    }

    return cnt.all { it <= 1 }
}